### DISCUSSION TITLE
Real Serial Printer on RS232-2

### DISCUSSION THREAD

**Discussion by Yakskiis** [thehighnibble/thehighnibble.github.io#71] [IMSAI8080esp replica]
I am trying to use a real printer with the IMSAI8080esp replica, a HP 2225D.

Is there a way to bind UL1 to output on RS232-2?

I know by default it is at UC1, and I can output the terminal output to the printer and even pip to uc1 to print, but I am trying to print from Wordstar and it looks at UL1 I think.

Even QT term outputs to the printer correctly, so I assume it looks for UC1 as well.

Or am i going the totally wrong direction to do this?

**Comment by udo-munk** [comment]
You can assign the printer to the second serial port with stat, like this:

A>stat val:

Temp R/O Disk: d:=R/O
Set Indicator: d:filename.typ $R/O $R/W $SYS $DIR
Disk Status  : DSK: d:DSK:
User Status  : USR:
Iobyte Assign:
CON: = TTY: CRT: BAT: UC1:
RDR: = TTY: PTR: UR1: UR2:
PUN: = TTY: PTP: UP1: UP2:
LST: = TTY: CRT: LPT: UL1:

A>stat lst:=ul1:

Wordstar (or any program doing output to a printer) outputs to LST: and with stat you route that to some physical output device, like UL1:. If you route LST: to TTY: for example, a program will print to the TTY console on the first serial port.
